src/averageAges.js

```diff
@@ -55,7 +67,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const men = people.filter(person => person.sex === 'm');
+  const children = onlyWithSon
+    ? men
+    : people;
+
+  const mothers = children
+    .map(child => people.find(person => child.mother === person.name));
+
+  const differenceBetweenAges = mothers.map((person, index) => {
+    const momAge = person === undefined
+      ? undefined
+      : children[index].born - person.born;
```

Comment:
Try to use some and filter methods to find mothers
You can filter the array of children and use some to get kids with mother